app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/CharecterView.kt

```diff
@@ -50,4 +67,44 @@ class CharecterView(activity: MainActivity) {
         activityRef.get()!!.progressBar.visibility = View.VISIBLE
 
     }
+
+
+
+    fun showCharacterInformation(character: Character) {
+        val dialogFragment = CharacterDialogFragment()
+        val args = Bundle()
+        args.putParcelable(EXTRA_CHARACTER, character)
+        dialogFragment.arguments = args
+        dialogFragment.show(activityRef.get()?.supportFragmentManager, "txn_tag")
+    }
+
+    class CharacterDialogFragment() : DialogFragment() {
+
+        private lateinit var character: Character
+
+
+        override fun onCreate(savedInstanceState: Bundle?) {
+            super.onCreate(savedInstanceState)
+        }
+
+        override fun onStart() {
+            super.onStart()
+            val d: Dialog? = dialog
+            if (d != null) {
+                val width = ViewGroup.LayoutParams.MATCH_PARENT
+                val height = ViewGroup.LayoutParams.MATCH_PARENT
+                d.window?.setLayout(width, height)
```

Comment:
why do you set height and width programmatically?

Reply:
Because i couldnt set in other way

Reply:
Can you share with us the example that you use?

Reply:
In next commit you will not see this. I implement the dialog fragment another way like you recommend me.